netbox/resource_netbox_config_context.go

```diff
@@ -0,0 +1,514 @@
+package netbox
+
+import (
+	"strconv"
+
+	"github.com/fbreckle/go-netbox/netbox/client"
+	"github.com/fbreckle/go-netbox/netbox/client/extras"
+	"github.com/fbreckle/go-netbox/netbox/models"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	// "github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func resourceNetboxConfigContext() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceNetboxConfigContextCreate,
+		Read:   resourceNetboxConfigContextRead,
+		Update: resourceNetboxConfigContextUpdate,
+		Delete: resourceNetboxConfigContextDelete,
+
+		Description: `:meta:subcategory:Extras:From the [official documentation](https://docs.netbox.dev/en/stable/models/extras/configcontext/):
+
+> Context data is made available to devices and/or virtual machines based on their relationships to other objects in NetBox. For example, context data can be associated only with devices assigned to a particular site, or only to virtual machines in a certain cluster.`,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"weight": &schema.Schema{
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  1000,
+			},
+			"data": &schema.Schema{
+				Type:     schema.TypeMap,
+				Required: true,
+				//				ValidateFunc: validation.StringLenBetween(0, 30),
```

Comment:
There is a json validation function at https://pkg.go.dev/github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/validation#StringIsJSON

Is the Map type powerful enough to cover all json usecases? If not, we should use a json and have the user generate it as needed, then just validate that its valid json.

Reply:
It turns out Map isn't up to the job, you're correct. Do you see this as being a string and end up like below?
```
resource "netbox_config_context" "test" {
  name = "test"
  data = "{\"key\": \"value\"}"
}
```
My worry would be string escaping, but I'll give it a go tomorrow and see what it looks like

Reply:
well, the users would probably just use the `jsonencode` function. The examples should eventually reflect that. [hererdocs](https://developer.hashicorp.com/terraform/language/expressions/strings#heredoc-strings) are an option and are widely used, but are advised against by hashicorp [link](https://developer.hashicorp.com/terraform/language/expressions/strings#generating-json-or-yaml).